sdks/java/io/kinesis/src/main/java/org/apache/beam/sdk/io/kinesis/ShardReadersPool.java

```diff
@@ -69,25 +76,37 @@ void start() throws TransientKinesisException {
       shardsMap.put(checkpoint.getShardId(), createShardIterator(kinesis, checkpoint));
     }
     shardIteratorsMap = shardsMap.build();
-    executorService = Executors.newFixedThreadPool(shardIteratorsMap.size());
-    recordsQueue = new LinkedBlockingQueue<>(queueCapacityPerShard * shardIteratorsMap.size());
-    for (final ShardRecordsIterator shardRecordsIterator : shardIteratorsMap.values()) {
-      executorService.submit(new Runnable() {
-
-        @Override
-        public void run() {
-          readLoop(shardRecordsIterator);
-        }
-      });
+    if (!shardIteratorsMap.isEmpty()) {
```

Comment:
When shard iterators map can be empty?

Reply:
It can happen in a situation when two shards that were in separate ShardReadersPools are merged into one. Then only one ShardReadersPool will pick up the resulting shard and the other one will not. If it was the only shard that pool was reading, then it should stop doing anything. This check is useful when restoring from the empty checkpoint. We’ve seen such situation a while ago.